eliot/_action.py

```diff
@@ -52,6 +52,10 @@ class _ExecutionContext(threading.local):
     """
 
     def __init__(self):
+        """
+        Every time an attribute is looked up from a new thread, this will be
+        called again for that thread.
+        """
```

Comment:
Oof.  That seems like a pretty important piece of information.  At the same time, the `__init__` docstring seems like a weird place for it - I guess because the information is mostly of interest to maintainers of `_ExecutionContext`, not users.  Maybe this is just the nature of `threading.local`... but maybe the code would be less surprising if `_ExecutionContext` has-a `threading.local` instead of is-a `threading.local`?

Reply:
Below, where GitHub won't let me leave a comment, `_get_stack` is documented as "Get the stack for the current asyncio Task."  I guess that could be updated since (even before this change) it's not necessarily the case that the stack belongs to an asyncio Task at this level.